colserde: don't generate fixed width binary data in tests

asubiotto opened the pull request:
Arrow was changed to represent fixed width binary data using two buffers (minus
the offsets buffer), which would break our assumption that all types.Bytes data
is represented using a third buffer (offsets). Since fixed width binary
generation was added for more test coverage but doesn't translate to a real use
case, we disable the fixed width binary data generation (we still keep
variable-width bytes and string generation).

Release note: None (testing fix)

Fixes #49267

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/49269)
<!-- Reviewable:end -->

yuzefovich reviewed (commented):
> Reviewed 1 of 1 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/49269)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @asubiotto)

___
*[pkg/col/colserde/record_batch_test.go, line 136 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/49269#-M7hiw8W1JbbUgR2zI0P:-M7hiw8W1JbbUgR2zI0Q:b-ywx0c1) ([raw file](https://github.com/cockroachdb/cockroach/blob/bbceab563eaf94117841797f58ee7670026d850f/pkg/col/colserde/record_batch_test.go#L136)):*
> ```Go
> 			}
> 			builder.(*array.BinaryBuilder).AppendValues(data, valid)
> 		case 2:
> ```

Should we delete or comment out this case then?


<!-- Sent from Reviewable.io -->

Commit:
colserde: don't generate fixed width binary data in tests

Arrow was changed to represent fixed width binary data using two buffers (minus
the offsets buffer), which would break our assumption that all types.Bytes data
is represented using a third buffer (offsets). Since fixed width binary
generation was added for more test coverage but doesn't translate to a real use
case, we disable the fixed width binary data generation (we still keep
variable-width bytes and string generation).

Release note: None (testing fix)

asubiotto reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/49269)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @yuzefovich)

___
*[pkg/col/colserde/record_batch_test.go, line 136 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/49269#-M7hiw8W1JbbUgR2zI0P:-M7hjU5ZBWe4gjpetffS:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/bbceab563eaf94117841797f58ee7670026d850f/pkg/col/colserde/record_batch_test.go#L136)):*
<details><summary><i>Previously, yuzefovich wrote…</i></summary><blockquote>

Should we delete or comment out this case then?
</blockquote></details>

Done.


<!-- Sent from Reviewable.io -->

yuzefovich reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 1 of 1 files at r2.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/49269#-:-M7hlPJL5hmtpBvH0qX3:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained



<!-- Sent from Reviewable.io -->

asubiotto commented:
TFTR

bors r=yuzefovich

yuzefovich reviewed (commented):
This makes me think that it might be worth implementing a special physical representation for `UUID` type that uses fixed width (16) binary format.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/49269#-:-M7hlV7b6q8PIr6wmTbC:b-6g4p9o)* status: :shipit: complete! 1 of 0 LGTMs obtained



<!-- Sent from Reviewable.io -->
